dvc/output.py

```diff
@@ -325,7 +325,9 @@ def __init__(  # noqa: C901
         meta = Meta.from_dict(info or {})
         # NOTE: when version_aware is not passed into get_cloud_fs, it will be
         # set based on whether or not path is versioned
-        fs_kwargs = {"version_aware": True} if meta.version_id else {}
+        fs_kwargs = {}
```

Comment:
> Thinking about it more, it should be specific to `Dependency.__init__()`. For outputs it doesn't work because `out.fs` is local/repo FS (and should not be version aware) even when we have `files` (from pushing `out` to a remote)

_Originally posted by @pmrowla in https://github.com/iterative/dvc/issues/8853#issuecomment-1399196513_

In the case of filesystems that don't explicitly handle `version_aware` in `prepare_credentials` (like local, repo), the flag is ignored so it is safe to do it here. 
It doesn't feel too clean to rely on the fs ignoring the flag, though

Reply:
This will probably need to be revisited at some point given that there's already open discussion about whether or not dvc should drop .dir usage in favor of files in all cases (not just versioned ones), but it works for now.